scripts/rules.go

```diff
@@ -134,6 +134,34 @@ func databaseImport(m dsl.Matcher) {
 		Where(m.File().PkgPath.Matches("github.com/coder/coder/v2/codersdk"))
 }
 
+// publishInTransaction detects calls to Publish inside database transactions
+// which can lead to connection starvation.
+//
+//nolint:unused,deadcode,varnamelen
+func publishInTransaction(m dsl.Matcher) {
+	m.Import("github.com/coder/coder/v2/coderd/database/pubsub")
+
+	// Match direct calls to the Publish method of a pubsub instance inside InTx
+	m.Match(`
+		$x.InTx(func($y) error {
+			$*_
+			$_ = $ps.Publish($evt, $msg)
```

Comment:
IIRC you need to also match `:=` separately.

Reply:
Yea, it's incredibly frustrating how many permutations can exist. Good callout of the inline `if` one.